src/test/rustdoc-ui/infinite-recursive-type-impl-trait-return.rs

```diff
@@ -1,12 +1,10 @@
-// check-pass
```

Comment:
(@jyn514 might have more insight whether this specific test came from some real-world code, or is ok to regress).

Reply:
Erf, that'd be pretty bad if some docs rely on that. At the very least, we should run a crater run to see what the impact is. If there is none, maybe we can consider changing this behaviour? I'd really prefer to avoid rewriting the `hir::Visitor` if possible.

Reply:
Actually, I should have looked at the original test impl. It was added as a failing test then got changed to `check-pass` in <https://github.com/rust-lang/rust/pull/102890>. So seems likely that it's not relied on in the wild as it isn't yet in a stable release.

Reply:
But your concerns remain completely valid. I think a crater run should definitely be run if we agree on this PR before r+ing it.

Reply:
So what should we do here? Is it ok and then we run a crater or this approach isn't the right one and we close this PR?

Reply:
> Actually, I should have looked at the original test impl. It was added as a failing test then got changed to check-pass in https://github.com/rust-lang/rust/pull/102890. So seems likely that it's not relied on in the wild as it isn't yet in a stable release.

That sounds right - the original test was just to make sure rustdoc didn't ICE.

Reply:
Ok, I'll start a crater run soon then.